File: backend/app/schemas/ai.py
from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel


class AIInsightOut(BaseModel):
    id: str
    dataset_id: str
    dataset_version_id: Optional[str]
    insight_type: str
    content: Optional[str]
    confidence_score: Optional[float]
    metadata_json: Optional[Dict[str, Any]]
    created_at: datetime

    model_config = {"from_attributes": True}


class AnalyzeRequest(BaseModel):
    dataset_id: str
    dataset_version_id: Optional[str] = None

File: frontend/src/lib/api/ai.ts
import { apiClient, ApiResponse } from "./client";

export interface AIInsight {
  id: string;
  dataset_id: string;
  dataset_version_id: string | null;
  insight_type:
    | "task_detection"
    | "cleaning_suggestion"
    | "model_recommendation"
    | "data_quality_warning";
  content: string | null;
  confidence_score: number | null;
  metadata_json: Record<string, unknown> | null;
  created_at: string;
}

export interface TranslateTrainingResult {
  task_type: string;
  model_type: string;
  target_column: string;
  confidence: number;
}

export const aiApi = {
  triggerAnalysis: (datasetId: string, datasetVersionId?: string) =>
    apiClient.post<ApiResponse<{ job_id: string }>>("/ai/analyze", {
      dataset_id: datasetId,
      dataset_version_id: datasetVersionId ?? null,
    }),

  getInsights: (datasetId: string) =>
    apiClient.get<ApiResponse<AIInsight[]>>(`/ai/insights/${datasetId}`),

  translateTrainingCommand: (body: { dataset_id: string; command: string }) =>
    apiClient.post<ApiResponse<TranslateTrainingResult>>("/ai/translate-training-command", body),

  recommendPipeline: (body: { dataset_id: string; project_id: string }) =>
    apiClient.post<ApiResponse<{ pipeline_id: string; pipeline_name: string }>>(
      "/ai/recommend-pipeline",
      body
    ),
};
